File: csharp/coupon/InsertCoupons.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using com.ultracart.admin.v2.Api;
using com.ultracart.admin.v2.Model;

namespace SdkSample.coupon
{
    public class InsertCoupons
    {
        public static void Execute()
        {
            Console.WriteLine("--- " + MethodBase.GetCurrentMethod()?.DeclaringType?.Name + " ---");
            try
            {
                // Create coupon API instance using API key
                CouponApi couponApi = new CouponApi(Constants.ApiKey);
                
                Coupon coupon1 = new Coupon();
                coupon1.MerchantCode = "PennyOff";
                coupon1.Description ="Test Coupon for InsertCoupons sample";
                coupon1.AmountOffSubtotal = new CouponAmountOffSubtotal(); // see InsertCoupon for examples of types
                coupon1.AmountOffSubtotal.DiscountAmount = 0.01m;

                Coupon coupon2 = new Coupon();
                coupon2.MerchantCode = "TwoPenniesOff";
                coupon2.Description ="Test Coupon for InsertCoupons sample";
                coupon2.AmountOffSubtotal = new CouponAmountOffSubtotal(); // see InsertCoupon for examples of types
                coupon2.AmountOffSubtotal.DiscountAmount = 0.02m;
                
                CouponsRequest couponsRequest = new CouponsRequest();
                couponsRequest.Coupons = new List<Coupon> { coupon1, coupon2 };
                var apiResponse = couponApi.InsertCoupons(couponsRequest);
                
                Console.WriteLine(apiResponse);

                foreach (Coupon coupon in apiResponse.Coupons)
                {
                    Console.WriteLine($"Deleting newly created coupon (Coupon OID {coupon.CouponOid}) to clean up.");
                    couponApi.DeleteCoupon(coupon.CouponOid);
                }
                
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
                Console.WriteLine(ex.StackTrace);
            }
        }
    }
}

File: javascript/coupon/insertCoupons.js
// Import API and UltraCart types
import { couponApi } from '../api.js';

// Namespace-like structure using a class
export class InsertCoupons {
  static async execute() {
    console.log(`--- InsertCoupons ---`);
    try {
      // Create coupon objects
      const coupon1 = {
        merchant_code: "PennyOff",
        description: "Test Coupon for InsertCoupons sample",
        amount_off_subtotal: {
          discount_amount: 0.01, // Decimal becomes number in TypeScript
        }, // See InsertCoupon for examples of types
      };

      const coupon2 = {
        merchant_code: "TwoPenniesOff",
        description: "Test Coupon for InsertCoupons sample",
        amount_off_subtotal: {
          discount_amount: 0.02, // Decimal becomes number in TypeScript
        }, // See InsertCoupon for examples of types
      };

      // Create CouponsRequest object
      const couponsRequest = {
        coupons: [coupon1, coupon2],
      };

      const apiResponse = await new Promise((resolve, reject) => {
        couponApi.insertCoupons(couponsRequest, function (error, data, response) {
          if (error) {
            reject(error);
          } else {
            resolve(data, response);
          }
        });
      });

      console.log(apiResponse);

      // Clean up: delete newly created coupons
      if (apiResponse.coupons) {
        for (const coupon of apiResponse.coupons) {
          console.log(`Deleting newly created coupon (Coupon OID ${coupon.coupon_oid}) to clean up.`);
          if (coupon.coupon_oid) {
            await new Promise((resolve, reject) => {
              couponApi.deleteCoupon(coupon.coupon_oid, function (error, data, response) {
                if (error) {
                  reject(error);
                } else {
                  resolve(data, response);
                }
              });
            });
          }
        }
      }
    } catch (ex) {
      console.log(`Error: ${ex.message}`);
      console.log(ex.stack);
    }
  }
}

// Example usage (optional, remove if not needed)
// InsertCoupons.execute().catch(console.error);